tools/deno_http_proxy.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import {
+  serve,
+  ServerRequest
+} from "../js/deps/https/deno.land/std/http/server.ts";
+
+const addr = Deno.args[1] || "127.0.0.1:4500";
+const originAddr = Deno.args[2] || "127.0.0.1:4501";
+const server = serve(addr);
+
+async function main(): Promise<void> {
+  console.log(`http://${addr}/`);
+  for await (const req of server) {
+    proxyRequest(req);
+  }
+}
+
+async function proxyRequest(req: ServerRequest) {
+  const url = `http://${originAddr}${req.url}`;
+  const resp = await fetch(url, {
+    method: req.method,
+    headers: req.headers
+  });
+  req.respond({
+    headers: resp.headers,
+    status: resp.status,
+    body: resp.body
+  });
```

Comment:
It's awesome how simple this example is.
It would be cool if we could do
```ts
req.respond(resp)
```
Is that meaningful in the browser?

Reply:
Yeah that should work! Also `fetch` really is great.